addons/account/views/report_invoice.xml

```diff
@@ -194,6 +231,7 @@
         <template id="report_invoice">
             <t t-call="web.html_container">
                 <t t-foreach="docs" t-as="o">
+                    <t t-set="html_view" t-value="o.env.context.get('html_view')"/>
```

Comment:
I really don't like tweaking the view depending on some context parameter. You should keep things explicit. What is t he purpose of that key ? Why is it needed ?

Reply:
@tde-banana-odoo 
we are using existing invoice report for two purpose 
1) print and download(PDF view)
2) display on portal (HTML view)

Currently while printing report(from frontend and backend) it will takes default xs view, xs class is applied and viewport size not applied on pdf. 

so if we add xs class directly on report XML it will effect main report layout (while print pdf) so only for display purpose in mobile (for responsive)  we added xs classes based on context parameter.

is it Okey for you ??